vllm/worker/model_runner.py

```
@@ -1818,6 +1819,9 @@ def need_recv_kv(self, model_input, kv_caches) -> bool:
         if self.vllm_config.kv_transfer_config is None:
             return False
 
+        if model_input.attn_metadata is None:
+            return False
+
```

This is needed to fix the complain from pre-commit 
```
Error: vllm/worker/model_runner.py:1822: error: Item "None" of "Any | None" has no attribute "prefill_metadata"  [union-attr]
```

---

Before this PR, `model_input.attn_metadata.prefill_metadata` would cause an error if `model_input.attn_metadata is None`. So that we don't change the behaviour, could we raise an error here instead? (Same below)